apps/rush-lib/src/scripts/install-run-common.ts

```diff
@@ -0,0 +1,299 @@
+          }
+        }
+
+        if (!lineShouldBeTrimmed) {
+          resultLines.push(line);
+        }
+      }
+
+      fs.writeFileSync(packageInstallNpmrcPath, resultLines.join(os.EOL));
+    } catch (e) {
+      throw new Error(`Error reading or writing .npmrc file: ${e}`);
+    }
+  }
+}
+
+/**
+ * Detects if the package in the specified directory is installed
+ */
+function isPackageAlreadyInstalled(packageInstallFolder: string): boolean {
+  try {
+    const flagFilePath: string = path.join(packageInstallFolder, INSTALLED_FLAG_FILENAME);
+    if (!fs.existsSync(flagFilePath)) {
+      return false;
+    }
+
+    const fileContents: string = fs.readFileSync(flagFilePath).toString();
+    return fileContents.trim() === process.version;
+  } catch (e) {
+    return false;
+  }
+}
+
+/**
+ * Removes the installed.flag file and the node_modules folder under the specified folder path.
+ */
+function cleanInstallFolder(rushCommonFolder: string, packageInstallFolder: string): void {
+  try {
+    const flagFile: string = path.resolve(packageInstallFolder, INSTALLED_FLAG_FILENAME);
+    if (fs.existsSync(flagFile)) {
+      fs.unlinkSync(flagFile);
+    }
+
+    const nodeModulesFolder: string = path.resolve(packageInstallFolder, NODE_MODULES_FOLDER_NAME);
+    if (fs.existsSync(nodeModulesFolder)) {
+      const rushRecyclerFolder: string = ensureAndResolveFolder(
+        rushCommonFolder,
+        'temp',
+        'rush-recycler',
+        Date.now().toString()
+      );
+      fs.renameSync(nodeModulesFolder, rushRecyclerFolder);
+    }
+  } catch (e) {
+    throw new Error(`Error cleaning the package install folder (${packageInstallFolder}): ${e}`);
+  }
+}
+
+function createPackageJson(packageInstallFolder: string, name: string, version: string): void {
+  try {
+    const packageJsonContents: IPackageJson = {
+      'name': 'ci-rush',
+      'version': '0.0.0',
+      'dependencies': {
+        [name]: version
+      },
+      'description': 'DON\'T WARN',
+      'repository': 'DON\'T WARN',
+      'license': 'MIT'
+    };
+
+    const packageJsonPath: string = path.join(packageInstallFolder, PACKAGE_JSON_FILENAME);
+    fs.writeFileSync(packageJsonPath, JSON.stringify(packageJsonContents, undefined, 2));
+  } catch (e) {
+    throw new Error(`Unable to create package.json: ${e}`);
+  }
+}
+
+/**
+ * Run "npm install" in the package install folder.
+ */
+function installPackage(packageInstallFolder: string, name: string, version: string): void {
+  try {
+    console.log(`Installing ${name}...`);
+    const npmPath: string = getNpmPath();
+    childProcess.execSync(`"${npmPath}" install`, { cwd: packageInstallFolder });
+    console.log(`Successfully installed ${name}@${version}`);
+  } catch (e) {
+    throw new Error(`Unable to install package: ${e}`);
+  }
+}
+
+/**
+ * Try to resolve the specified binary in an installed package.
+ */
+function findBinPath(packageInstallFolder: string, name: string, binName: string): string {
```

Comment:
>findBinPath [](start = 9, length = 11)

NPM will create a "node_modules/.bin" folder with  a script that we can spawn, so there's no need to parse the package.json file here.

Instead I just use spawn() to invoke the binary.  On win32 only, you need to append ".cmd" because there is no shebang concept on Windows